activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/JsonSchemaModelValidator.java

```diff
@@ -41,111 +45,145 @@
  */
 public abstract class JsonSchemaModelValidator implements ModelValidator {
 
-    private final Logger log = LoggerFactory.getLogger(JsonSchemaModelValidator.class);
-
-    protected abstract SchemaLoader schemaLoader();
-
-    @Override
-    public void validate(byte[] bytes,
-                         ValidationContext validationContext) {
-        JSONObject processExtensionJson = null;
-        try {
-            log.debug("Validating json model content: " + new String(bytes));
-            processExtensionJson = new JSONObject(new JSONTokener(new String(bytes)));
-            schemaLoader()
-                    .load()
-                    .build()
-                    .validate(processExtensionJson);
-        } catch (JSONException jsonException) {
-            log.debug("Syntactic model JSON validation errors encountered",
-                      jsonException);
-            throw new SyntacticModelValidationException(jsonException);
-        } catch (ValidationException validationException) {
-            log.debug("Semantic model validation errors encountered: " + validationException.toJSON(),
-                      validationException);
-            throw new SemanticModelValidationException(validationException.getMessage(),
-                                                       getValidationErrors(validationException, processExtensionJson));
-        }
+  private final Logger log = LoggerFactory.getLogger(JsonSchemaModelValidator.class);
+
+  protected abstract SchemaLoader schemaLoader();
+
+  @Override
+  public void validate(byte[] modelContent,
+                       ValidationContext validationContext) {
+    internalValidation(modelContent, true);
```

Comment:
We should probably remove `shouldThrowError` and avoid having a method which behavior changes basing on a boolean condition. We could instead decide inside `validate` if we have to throw an exception or not.
I would suggest to check if the collection contains a `SyntacticModelError` and throw a `SyntacticModelValidationException` otherwise if there are other errors to throw a `SemanticModelValidationException` and so on.
Since this logic seems common, we could probably also create an utility class for this kind of behavior to reuse it where it is required.

Reply:
There are two scenarios where caller expect exception and second when expect collection that might contain validation errors, thats why there is a boolean. 
So how you would like to decide to throw or not? See only option where code will always collect and then throw or not